show list of errors on inconsistent tests

bobholt opened the pull request:
This is a re-implementation of https://github.com/w3c/web-platform-tests/pull/4916 with the changes requested in the review after that PR was closed.

@jgraham: I'm happy to continue the review here.

Commit:
show list of errors on inconsistent tests

w3c-bots commented:
[View the complete job log.](https://travis-ci.org/w3c/web-platform-tests/jobs/204644352)

# Firefox (nightly channel) #
Testing web-platform-tests at revision cc1ba0d8a381b5f73c5dd069121ff09f7afbb9b5
Using browser at version BuildID 20170222110216; SourceStamp 9a9db410f20781076424307265decbc5de1e94cc
Starting 10 test iterations
No tests run.

w3c-bots commented:
[View the complete job log.](https://travis-ci.org/w3c/web-platform-tests/jobs/204644355)

# Chrome (unstable channel) #
Testing web-platform-tests at revision cc1ba0d8a381b5f73c5dd069121ff09f7afbb9b5
Using browser at version 58.0.3018.3 dev
Starting 10 test iterations
No tests run.

jgraham commented on check_stability.py:
```diff
@@ -487,13 +487,44 @@ class LogHandler(reader.LogHandler):
 
         def test_status(self, data):
-            self.results[data["test"]][data.get("subtest")][data["status"]] += 1
+            subtest = self.find_or_create_subtest(data)
```
I'm quite tempted to inline this since it should always exist and function calls are rather expensive. It probably doesn't matter though, so up to you.

bobholt replied:
In editing this code yesterday, it was easier to understand and make the parallel changes to `test` and `subtest` functionality with it broken out, so I'd like to keep it as-is for maintainability.

jgraham commented on check_stability.py:
```diff
@@ -522,10 +553,10 @@ def process_results(log, iterations):
-        for subtest, result in test_results.iteritems():
-            if is_inconsistent(result, iterations):
-                inconsistent.append((test, subtest, result))
+    for test_name, test in results.items():
```
Use `iteritems` here and below.

Also, since you moved the test status up into the test object rather than making it a "subtest" with no name, this check misses tests that have an unstable overall status. So you either need to revert that change or fix up here (and maybe elsewhere) that assumes the parent test is represented as a subtest with `None` as the name.

jgraham commented on check_stability.py:
```diff
@@ -592,22 +628,24 @@ def write_results(results, iterations, comment_pr):
                                                           else "test"))
 
-    for test, test_results in results.iteritems():
+    for test_name, test in results.items():
```
`iteritems()`

bobholt commented:
@jgraham I believe I've addressed your comments. I added a fixup commit for easier review of the change and an unstable test to prove to myself it still worked. I'll squash after you take a look.

jgraham reviewed (approved):
This looks OK to me now. Thanks for making the changes!

bobholt commented:
Thanks @jgraham. Just pushed a single commit. 

For posterity, the current unstable output looks like:
![raise_errors](https://cloud.githubusercontent.com/assets/94167/23306334/1e19b8c0-fa71-11e6-9c13-b141e418abb7.png)
